docs/policy/memtier.md

```diff
@@ -1,70 +1,314 @@
+    * eligible to run on `kube-reserved` CPU cores
+    * always run on shared CPU cores
+  - `low-priority`
+    * not eligible for extra optimizations
+    * always run on shared CPU cores
+  - `sub-core`
+    * not eligible for extra optimizations
+    * always run on shared CPU cores
+  - `mixed`
+    * by default eligible for exclusive and isolated allocation
+    * not eligible for either if `PreferSharedCPUs` is set to true
+    * not eligible for either if annotated to opt out from exclusive allocation
+    * not eligible for isolated allocation if annotated to opt out
+  - `multi-core`
+    * CPU request fractional (`(CPU request % 1000 milli-CPU) != CPU request`):
+      - by default not eligible for extra optimizations
+      - eligible for exclusive and isolated allocation if annotated to opt in
+    * CPU request not fractional:
+      - by default eligible for exclusive allocation
+      - by default not eligible for isolated allocation
+      - not eligible for exclusive allocation if annotated to opt out
+      - eligible for isolated allocation if annotated to opt in
+
+Eligibility for kube-reserved CPU core allocation should always be possible to
+honor. If this is not the case, it is probably due to an incorrect configuration
+which underdeclares `ReservedResources`. In that case, ordinary shared CPU cores
+will be used instead of kube-reserved ones.
+
+Eligibility for exclusive CPU allocation should always be possible to honor. If
+not, it is probably due to bugs in the resource accounting algorithm and should
+be reported as such.
+
+Eligibility for isolated core allocation is only honored if there are enough
+isolated cores available to fulfill the exclusive part of the container's CPU
+request with isolated cores alone. Otherwise ordinary CPUs will be allocated,
+by slicing them off for exclusive usage from the shared subset of CPU cores in
+the container's assigned pool.
+
+Containers in the kube-system group are pinned to share all kube-reserved CPU
+cores. Containers in the low-priority or sub-core groups, and containers which
+are only eligible for shared CPU core allocation in the mixed and multi-core
+groups, are all pinned to run on the shared subset of CPU cores in the
+container's assigned pool. This shared subset can and usually does change
+dynamically as exclusive CPU cores are allocated and released in the pool.
+
+## Container CPU Allocation Preference Annotations
+
+Containers can be annotated to diverge from the default CPU allocation
+preferences the policy would otherwise apply to them. These annotations
+can be given both with per pod and per container resolution. If for any
+container both of these exist, the container-specific one takes precedence.
+
+### Shared, Exclusive, and Isolated CPU Preference
+
+A container can opt in to or opt out from shared CPU allocation using the
+following annotation.
+
+```yaml
+metadata:
+  annotations:
+    # opt in container $C1 to shared CPU core allocation
```

Comment:
In existing examples we don't use `$` as part of container names. Perhaps we could go here with `$`, too? (At least it made me wonder if one needs to prefix container names by `$` when using this notation.)

Reply:
Removed $ from all example container names.